src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9300,6 +9300,65 @@ namespace SurfaceGeometry {
         }
     }
 
+    struct PopCoincidentVertexReturn
+    {
+        // bool popNeeded;
+        double perimeter;
+        int poppedVertexPos = -1; // This is a STL vector position, 0-indexed
+        int keptVertexPos = -1;
+    };
+
+    PopCoincidentVertexReturn checkPopCoincidentVertex(const Array1D<Vector> &vertices)
+    {
+        constexpr double tolerance = 0.01;
+
+        size_t const nSides = vertices.size();
+
+        // Vector of distance from this vertex to the next one
+        std::vector<Real64> distances(nSides);
+        size_t index = 0;
+        double min_distance = std::numeric_limits<Real64>::max();
+        double perimeter = 0.0;
+        for (auto it = vertices.begin(); it != vertices.end(); ++it) {
+            auto itnext = std::next(it);
+            if (itnext == std::end(vertices)) {
+                itnext = std::begin(vertices);
+            }
+            const auto dist = distance(*it, *itnext);
+            distances[index++] = dist;
+            min_distance = std::min(min_distance, dist);
+            perimeter += dist;
+        }
+        if (min_distance >= tolerance) {
+            return {perimeter};
+        }
```

Comment:
Pass one: calculate all distances between vertex N and N+1.
Return early if none will need to be popped.